### PR TITLE
Only output Schema when the checks for required blocks are OK

### PR DESCRIPTION
**Pull request by johannadevos**
## Summary

This PR can be summarized in the following changelog entry:

* [@yoast/schema-blocks] Fixes a bug where Schema output would still be generated if all the required blocks had been valid before, even if they were no longer valid now.
* [@yoast/schema-blocks] Fixes a bug where Schema output would be generated even when no variation had been picked for the Location block.

## Relevant technical choices:

* If the Job Posting block overall isn't valid (`validation.result > BlockValidation.Valid`), we should dispatch a `null` Schema output to the store, to overwrite the Schema output that was potentially stored there before. 
* The Job location block template has multiple instructions that are validated. Therefore, these results should be combined in an 'overall' validation result by using `validateMany`. In order to do so, we first need to collect these results and flatten them into a list using `getAllDescendantIssues`.
* I created new values from `MissingBlock` in the `BlockValidation` enum: `MissingRecommendedBlock` and `MissingRequiredBlock`. This is because a missing recommended block should not make the overall validation result invalid, whereas a missing required block should. Now we can make that distinction.

## Test instructions

This PR can be tested by following these steps:

* Add a Job Posting block to your post. Don't fill in anything.
* Publish the post, inspect the Schema output, and see that the `JobPosting` graph piece is **absent**.
* Fill in the title and description, but don't choose a location in the location picker.
    * Move the focus to another block to make sure the validations are executed (should be fixed in https://yoast.atlassian.net/browse/P2-780).
    * Update the post, see that the `JobPosting` graph piece is still **absent**.
* Choose a location in the location picker (preferably the remote location, because that one is already filled in).
    * Move the focus to another block to make sure the validations are executed.
    * Update the post, see that the `JobPosting` graph piece is now **present**.
* Remove a recommended block. 
    * Move the focus to another block to make sure the validations are executed.
    * See an orange bullet for that recommended block appear in the analysis.
    * See that the analysis conclusion is still green.
    * Update the post, see that the `JobPosting` graph piece is still **present**.
* There is a known bug: if you delete the text in the remote location, the `JobPosting` graph piece will stay present but it should be absent. This is due to https://yoast.atlassian.net/browse/P2-807. 
* Empty the title and/or description (don't remove them).
    * Move the focus to another block to make sure the validations are executed.
    * Update the post, see that the `JobPosting` graph piece is now **absent**.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes https://yoast.atlassian.net/browse/P2-695

### CODE REVIEW

File: packages/schema-blocks/src/functions/presenters/SidebarWarningPresenter.ts [hunk 1]
```diff
@@ -2,8 +2,8 @@ import { getBlockType } from "../BlockHelper";
 import { select } from "@wordpress/data";
 import { __ } from "@wordpress/i18n";
 import { get } from "lodash";
-import { BlockValidationResult } from "../../core/validation";
-import { BlockValidation } from "../../core/validation";
+import { BlockValidation, BlockValidationResult } from "../../core/validation";
+import { getAllDescendantIssues } from "../validators/getAllDescendantIssues";
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/presenters/SidebarWarningPresenter.ts:6]: mooier om in `validators/index.ts` deze functie te importeren en exporteren, zodat je hier
```suggestion
import { getAllDescendantIssues } from "../validators";
```
kunt doen

File: packages/schema-blocks/src/functions/validators/getAllDescendantIssues.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { BlockValidationResult } from "../../core/validation";
```
Comment (increddibelly) [inline packages/schema-blocks/src/functions/validators/getAllDescendantIssues.ts:1]: zie comment op `SidebarWarningPresenter`, dit is mooier.

**Review by increddibelly — CHANGES_REQUESTED** [review]
even de import toevoegen aan index.ts, zie comment bij `SidebarWarningPresenter`

File: packages/schema-blocks/src/functions/presenters/SidebarWarningPresenter.ts (hunk 1 above)
Comment (johannadevos) [reply packages/schema-blocks/src/functions/presenters/SidebarWarningPresenter.ts]: Done

**Review by increddibelly — APPROVED** [review]

**Comment by andizer** [comment]
Acceptance 👍

**Pull request merged**